endpoints/openrtb2/auction.go

```diff
@@ -127,8 +127,17 @@ func (deps *endpointDeps) Auction(w http.ResponseWriter, r *http.Request, _ http
 	if req.App != nil {
 		labels.Source = pbsmetrics.DemandApp
 		labels.RType = pbsmetrics.ReqTypeORTB2App
-		if req.App.Publisher != nil && req.App.Publisher.ID != "" {
-			labels.PubID = req.App.Publisher.ID
+		if req.App.Publisher != nil {
+			if req.App.Publisher.ID != "" {
+				labels.PubID = req.App.Publisher.ID
+			}
+			var pubExt openrtb_ext.ExtPublisher
+			if req.App.Ext != nil {
+				err := json.Unmarshal(req.App.Ext, &pubExt)
+				if err == nil && pubExt.ParentAccount != nil {
+					labels.PubID = *pubExt.ParentAccount
+				}
+			}
```

Comment:
In the above, if we can get PubID from req.App.Ext, we don't need to examine req.App.Publisher or anything beneath it at all.

As above, one thing not clear to me is whether it is intentional that req.App.Publisher not be nil before choosing to examine req.App.Ext.  As written, even if req.App.Ext can produce a value for PubID, it would not do so unless req.App.Publisher is non-nil.  My comment in the preceding paragraph assumes we would want to assign PubID from req.App.Ext even if req.App.Publisher is nil.